core/spi/src/main/java/org/wildfly/swarm/spi/api/MarkerContainer.java

```diff
@@ -32,12 +32,19 @@
 
     @SuppressWarnings("unchecked")
     default T addMarker(String markerName) {
-        addAsManifestResource(new StringAsset("marker"), ArchivePaths.create(PATH_MARKERS, markerName));
-
+        addMarker(this, markerName);
         return (T) this;
     }
 
     default boolean hasMarker(String markerName) {
-        return contains(ArchivePaths.create("META-INF/markers", markerName));
+        return hasMarker(this, markerName);
+    }
+
+    static void addMarker(ManifestContainer<?> manifestContainer, String markerName) {
```

Comment:
The benefit is we can use these methods for archives which do not implement `MarkerContainer` but fulfill the requirements.

> Also, should the marker methods both use Archive and not ManifestContainer?

I'm not sure. We could change the `PATH_MARKERS` to `META-INF/markers` but then we wouldn't allow the `ManifestContainer` to adjust the final path (e.g. WAR might use `WEB-INF/classes/META-INF/markers` instead). But the truth is that `addMarker()` and `hasMarker()` are not symmetric now... Unfortunately, `ManifestContainer` does not declare anything like `getManifestPath()`.